src/map/party.cpp

```diff
@@ -116,6 +126,28 @@ void CParty::DisbandParty(bool playerInitiated)
     {
         m_PAlliance->removeParty(this);
     }
+    else
+    {
+        // Before removing members from the party, remove them from the treasure pool
+        std::vector<std::pair<CZone*, CTreasurePool&>> PTreasurePools;
+
+        for (const auto& member : members)
+        {
+            if (auto* PChar = dynamic_cast<CCharEntity*>(member))
+            {
+                auto& PTreasurePool = PChar->GetTreasurePool();
+                PTreasurePools.emplace_back(zoneutils::GetZone(PChar->getZone()), PTreasurePool);
+                PTreasurePool.DelMember(PChar);
+            }
+        }
+
+        // Force release of treasure pools, otherwise the leader would get reattached to the same pool
+        for (const auto& poolToBeReleased : PTreasurePools)
```

Comment:
You can do this decomposition in the if block: `if (const auto& [PZone, PTreasurePool] : ...`